### PR TITLE
Release process: prepare JavaScript and typings

### LINKED ISSUE #13950: Redesign the release process
### 🎯 Goals

- Automate the process so it can be done with as little interaction from the Releaser as possible.
- Make the process bulletproof so more people can trigger it. Thus: No need for special planning.
- To enable publishing nightly CKEditor 5 packages.

### 💡 Hints

- Instead of implementing big tasks that do everything and are hard to introduce a new step into it, let's create small utils that can be glued and extendable.
- We can drop the current tools and create new ones from scratch. The old tools can be used as inspiration.

### LINKED ISSUE #13970: Release process: Prepare JS for the `ckeditor5-*` packages
## 📝 Provide a description of the new feature

**🎯 The goal is to prepare CKEditor 5 feature packages to be ready to publish on npm.**

- This step should be executed in the release directory.
- We do not want to publish TypeScript sources. Hence, compile them to create JavaScript and declaration files.
- The `"main"` field in a `package.json` should point to a JavaScript file instead of TS.
- TypeScript understands the `"types"` property. Hence, let’s add it. It should point to an entry point of typings.
- Remove TypeScript sources from packages to enable preparing DLLs from the JavaScript code.
- Bonus point: Could we execute the preparation step concurrently to improve building time?
- Possible arguments:
    - a directory where to look for packages (string)
- Ideas:
	- Perhaps this script could land in `ckeditor5-dev-*`
	- It depends on the API. Could we reuse a core of the feature when we prepare DLLs? (#13971)

---

If you'd like to see this feature implemented, add a 👍 reaction to this post.

### PR DESCRIPTION
**Pull request by pomek**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/contributing/git-commit-message-convention.html))

Internal: Release process: prepare JavaScript and typings. Closes #13970.

---

### Additional information

- [x] Requires: https://github.com/ckeditor/ckeditor5-dev/pull/869.

### CODE REVIEW

File: scripts/release/prepare-packages.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+		packagesDirectory: 'release',
+		processDescription: 'Compiling TypeScript...',
+		signal: abortController.signal,
+		taskToExecute: compileTypeScriptCallback
+	} );
+
+	/**
+	 * @param {String} packagePath
+	 */
+	function compileTypeScriptCallback( packagePath ) {
```
Comment (martnpaneq) [inline scripts/release/prepare-packages.js:42]: I think this function should also be stored in our tools. Typescript compilation is going to be called by 'collaboration-features` and `ckeditor5-internal` packages too.

File: scripts/release/prepare-packages.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+// Windows does not understand CTRL+C attached to `process`.
+if ( process.platform === 'win32' ) {
+	const rl = readline.createInterface( {
+		input: process.stdin,
+		output: process.stdout
+	} );
+
+	rl.on( 'SIGINT', () => process.emit( 'SIGINT' ) );
+}
```
Comment (psmyrek) [inline scripts/release/prepare-packages.js:25]: This works weird on Windows:
- I have the impression that I have to press `CTRL+C` several times and wait a few seconds for the script to be terminated.
- Additionally, if there is no `release` directory, the scripts hangs:
   


The good news is we don't need this whole `if` at all 😄 

Question: whether we have the `readline.createInterface()` or not, terminated script throws this error. Is it expected?

obraz

Comment (pomek) [reply scripts/release/prepare-packages.js:25]: The unhandled rejection error should be fixed in https://github.com/ckeditor/ckeditor5-dev/pull/869/commits/828a7e214b9688f62650ea739b0b84c4099f4c25.

**Review by psmyrek — APPROVED** [review]
LGTM 🎉 

It works as expected when `@ckeditor/ckeditor5-dev-utils` and `@ckeditor/ckeditor5-dev-release-tools` are linked in CKE5 repository.

File: scripts/release/prepare-packages.js (hunk 2 above)
Comment (psmyrek) [reply scripts/release/prepare-packages.js]: Thanks, it works fine now.

**Pull request merged**